pkg/vmprovider/providers/vsphere/virtualmachine/backup.go

```diff
@@ -37,6 +48,11 @@ func BackupVirtualMachine(
 		vmCtx.Logger.Info("No VM bootstrap data is provided for backup")
 	}
 
+	vmDiskData, err := getEncodedVMDiskData(vmCtx.Context, vcVM)
+	if err != nil {
+		vmCtx.Logger.Error(err, "Failed to get encoded VM disk data")
```

Comment:
Missing `return err`? Or is this purposely keep going to use an empty value (I'd `vmDiskData = ""` here to make that really clear then)

Reply:
Added the missing `return err` statement. For now we only proceed with calling VM reconfiguration if all the required backup data is available. It may be changed as a best effort in the follow-up optimization.